datafusion/expr/src/type_coercion/aggregates.rs

```diff
@@ -163,7 +142,7 @@ pub fn check_arg_count(
     Ok(())
 }
 
-fn get_min_max_result_type(input_types: &[DataType]) -> Result<Vec<DataType>> {
+pub fn get_min_max_result_type(input_types: &[DataType]) -> Result<Vec<DataType>> {
```

Comment:
I think we could keep this private

Reply:
This was used from `coerce_types` and is now used from `coerce_types` within the UDF implementation I can move it over to https://github.com/apache/datafusion/pull/11013/files#diff-531ff673a571f9ab5fa3eaf24ea3d4d26385ba88cf33a1c47293d0a180c0d453 but the other are udfs (sum, etc) are still below in this file

Reply:
Implement the logic inside `coerce_types`, we don't even need this function anymore and test either